modules/nf-core/sourmash/gather/main.nf

```diff
@@ -0,0 +1,53 @@
+process SOURMASH_GATHER {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::sourmash=4.5.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/sourmash:4.5.0--hdfd78af_0':
+        'quay.io/biocontainers/sourmash:4.5.0--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(signature)
+    path(database)
+    val save_unassigned
+    val save_matches_sig
+    val save_prefetch
+    val save_prefetch_csv
+
+    output:
+    tuple val(meta), path('*.csv')           , optional:true, emit: result
+    tuple val(meta), path('*_unassigned.sig'), optional:true, emit: unassigned
+    tuple val(meta), path('*_matches.sig')   , optional:true, emit: matches
+    tuple val(meta), path('*_prefetch.sig')  , optional:true, emit: prefetch
```

Comment:
Here's the lines of code that control prefect save behavior:
https://github.com/sourmash-bio/sourmash/blob/569d212d65d20a4c94680dd451834046e825949a/src/sourmash/sourmash_args.py#L1188

Saving like this is going to emit some really large files...it might be worth saving the files as a `*.zip`, so they're compressed and packaged together. Zips are also correctly formatted to pass as databases to gather in the future

Reply:
Compressed files are preferred in nf-core so that's a great hint 🙂

Reply:
`sourmash` is great, it supports compression out of the box.
I've updated the module - now the signatures are compressed with zip ([as recommended](https://sourmash.readthedocs.io/en/latest/databases-advanced.html#storing-signatures-in-zip-files-the-recommended-format)) and csv files are gzipped.

As a side effect, we cannot check md5 sums of the output now (I think it's because ZIP stores timestamps of the files).

PS. currently `--output-unassigned` argument ignores zip-compression of signatures, but I've reported this issue to the developers (https://github.com/sourmash-bio/sourmash/issues/2337).

Reply:
Bummer about the timestamps. With the gzip tool that's easily avoided but I guess here we'd need a new version of sourmash to make that happen.